ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/ConnectionProxy.java

```diff
@@ -61,6 +72,12 @@ public Session createSession(boolean transacted, int acknowledgeMode) throws JMS
         return xaConnection.createSession(transacted, acknowledgeMode);
     }
 
+    /**
+     * Simply close the proxied connection if there is no active transaction. Or register a
+     * {@link ConnectionClosingSynchronization} if active transaction exists.
+     * 
+     * @throws JMSException if transaction status check, synchronization registration, or connection closing has failed.
```

Comment:
What is a "a failed transaction status check". The javadoc needs to say which status values result in an exception being thrown.